preprocess_era5_modellevels.py

```diff
@@ -0,0 +1,178 @@
+import os
+
+import pandas as pd
+import xarray as xr
+import yaml
+import numpy as np
+
+from preprocessing import (get_grid_info, get_stable_fluxes,
+                           get_vertical_transport)
+
+
+# Set constants
+g = 9.80665  # [m/s2]
+density_water = 1000  # [kg/m3]
+
+# Select model levels
+modellevels = np.append( np.arange(1,100,10), np.arange(101,138,1)) #[1,20,40,60,80,100,110,120,125,130,131,132,133,134,135,136,137]
+print('Number of model levels:',len(modellevels))
+
+# Read case configuration
+with open("cases/era5_2013.yaml") as f:
+    config = yaml.safe_load(f)
+
+def load_surface_data(variable, date):
+    """Load data for given variable and date."""
+    filename = f"FloodCase_201305_{variable}.nc"
+    filepath = os.path.join(config["input_folder"], filename)
+    da = xr.open_dataset(filepath)[variable]
+
+    # Include midnight of the next day (if available)
+    extra = date + pd.Timedelta(days=1)
+    return da.sel(time=slice(date, extra))
+
+def load_modellevel_data(variable, date):
+    """Load model level data for given variable and date."""
+    filename = f"FloodCase_201305_ml_{variable}.nc"
+    filepath = os.path.join(config["input_folder"], filename)
+    da = xr.open_dataset(filepath)[variable]
+
+    # Include midnight of the next day (if available)
+    extra = date + pd.Timedelta(days=1)
+    return da.sel(time=slice(date, extra)).sel(lev=modellevels)
+
+
+datelist = pd.date_range(
```

Comment:
"start_date" and "end_date" have been renamed to "preprocess_start_date" and "preprocess_end_date" in the config file.

Suggested change
datelist = pd.date_range(
    start=config["preprocess_start_date"], end=config["preprocess_end_date"], freq="d", inclusive="left"
)